src/main/java/Entities/NormalGroup.java

```diff
@@ -4,6 +4,7 @@
 
 public class NormalGroup implements Serializable, Group {
     private String groupLeader;
+
     private HashMap<String, String> groupMembers; //maps username to username,
```

Comment:
Why are we mapping username to username? Is there a reason it needs to be a hashmap or could we make it an arraylist instead? Also I noticed that groupMembers used to map usernames to user objects, are we not doing that anymore because of serialization or something?

Reply:
Leo said that Hashmap is faster running time than an array. Yes we can't do usernames to Users because it made Serialization break.